superset/utils/decorators.py

```diff
@@ -116,3 +118,20 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def logged_in_api(f: Callable) -> Callable:
+    """
+        Use this decorator to require a user to be logged to use your API methods.
+
+        If the user is not logged in, this will return an HTTP 401.
+    """
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if not g.user.get_id():
```

Comment:
I think the check should be `if not g.user` as this would be `None` if there was no user logged in. This pattern is used [elsewhere](https://github.com/apache/incubator-superset/blob/f7812a3f27304f6ccb5932075891f162264cd88b/superset/views/core.py#L2661).

Reply:
turns out `g.user` exist when not logged in; it's an AnonymousUser. I think that testing for `.is_active` is best (as `is_active` also checks `is_authenticated`)